test/e2e/e2e-cli.sh

```diff
@@ -1,21 +1,26 @@
 #!/bin/sh
 
-npm run build
-
 WORK_DIR="./local/cli"
 CONFIG_FILE="./config-cli-e2e.json"
 
+npm ci #Install all dependencies to allow for building
+npm run build
+npm ci --omit=dev #Remove dev dependencies
+
+TARBALL_PATH=$(npm pack)
+npm install -g $TARBALL_PATH
```

Comment:
This is the bulk of the PR. We're basically simulating the publish process via `npm pack` and then simulating a user installing the package on their machine by installing the generated tarball.